packages/repository/src/__tests__/unit/repositories/constraint-utils.unit.ts

```diff
@@ -95,16 +95,50 @@ describe('constraint utility functions', () => {
       }).to.throwError(/Property "id" cannot be changed!/);
     });
 
+    it('allows constrained fields with the same values', () => {
+      const input = new Order({id: 2, description: 'order 1'});
+      const constraint: Partial<Order> = {id: 2};
+      const result = constrainDataObject(input, constraint);
+      expect(result).to.deepEqual(
+        new Order({
+          id: 2,
+          description: 'order 1',
+        }),
+      );
+    });
+  });
+
+  describe('constrainDataObjects', () => {
     it('constrains array of data objects', () => {
       const input = [
-        new Order({id: 1, description: 'order 1'}),
-        new Order({id: 2, description: 'order 2'}),
+        new Order({description: 'order 1'}),
+        new Order({description: 'order 2'}),
       ];
       const constraint: Partial<Order> = {id: 3};
       const result = constrainDataObjects(input, constraint);
       expect(result[0]).to.containDeep(Object.assign({}, input[0], constraint));
       expect(result[1]).to.containDeep(Object.assign({}, input[1], constraint));
     });
+
+    it('throws error when the query changes field in constrain', () => {
```

Comment:
Please change `constrain` to `constraint` (noun).

Reply:
Will fix here and in the original test that I copied from too.